language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -421,10 +525,155 @@ genModuleRef (pkgRef, modName) = case pkgRef of
 onHead :: (a -> a) -> [a] -> [a]
 onHead f = \case
     [] -> []
-    x:xs -> f x:xs
+    x : xs -> f x : xs
 
 onLast :: (a -> a) -> [a] -> [a]
 onLast f = \case
     [] -> []
     [l] -> [f l]
     x : xs -> x : onLast f xs
+
+writeTsConfig :: FilePath -> IO ()
+writeTsConfig dir = writeFileUTF8 (dir </> "tsconfig.json") $ unlines
+    [ "{"
+    , "  \"compilerOptions\": {"
+    , "    \"target\": \"es5\","
+    , "    \"lib\": ["
+    , "      \"dom\","
+    , "      \"es2015\""
+    , "     ],"
+    , "    \"strict\": true,"
+    , "    \"noUnusedLocals\": true,"
+    , "    \"noUnusedParameters\": false,"
+    , "    \"noImplicitReturns\": true,"
+    , "    \"noFallthroughCasesInSwitch\": true,"
+    , "    \"outDir\": \"lib\","
+    , "    \"module\": \"commonjs\","
+    , "    \"declaration\": true,"
+    , "    \"sourceMap\": true"
+    , "    },"
```

Comment:
That's indented too deep:
```suggestion
    , "  },"
```

Reply:
Noted. Will fix.

Reply:
Fixed in https://github.com/digital-asset/daml/pull/4818.